crypto/address.go

```diff
@@ -186,3 +187,14 @@ func NewContractAddress(caller Address, nonce []byte) (newAddr Address) {
 	copy(newAddr[:], hasher.Sum(nil))
 	return
 }
+
+func NewContractAddress2(caller Address, nonce []byte, salt [32]byte, initcode []byte) (newAddr Address) {
+	// sha3(0xff ++ caller.Address() ++ salt ++ sha3(init_code))[12:]
+	temp := make([]byte, 1+20+32+32)
+	temp[0] = 0xFF
+	copy(temp[1:], caller[:])
+	copy(temp[22:], salt[:])
+	copy(temp[53:], sha3.Sha3(initcode))
+	copy(newAddr[:], sha3.Sha3(temp)[12:])
```

Comment:
If I'm being a stickler I'd rather do the arithmetic in terms of constants too - all compile time anyway and a bit clearer/less error prone